aws-amplify/amplify-js
# [Delivers #163486970] Refactor API category and split into Rest and GraphQL modules

*Issue #:* https://github.com/aws-amplify/amplify-js/issues/3365

*Description of changes:*
Split API package into two packages, RestAPI and GraphQLAPI to support modular imports in current Amplify version (backwards compatible way). 

Tested following imports
* import { API } from 'aws-amplify'; // Imports the whole library, not recommended
* import { API } from '@aws-amplify/api'; // imports all API features including grapqhQL and Rest
* import API from '@aws-amplify/api'; // Same as above
* import { RestAPI } from '@aws-amplify/api-rest'; // Only imports Rest API functionality
* import { GraphQLAPI } from '@aws-amplify/api-graphql'; // imports GrapqhQL API functionality (including Rest api since that is what is used under the hood to make graphQL calls)


By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
